### ISSUE TITLE
Ensure function_exists() and class_exists() calls check the correct symbol name

### ISSUE THREAD

**Issue by johnbillion** [squizlabs/PHP_CodeSniffer#1744]
When conditionally declaring a function or class (ie. immediately following a function-not-exists or class-not-exists condition), it would be great if PHPCS triggered an error if the symbol names don't match. Examples of incorrect code:

```php
if ( ! function_exists( 'foo' ) ) {
	function bar() {
	}
}
```

```php
if ( ! class_exists( 'FooClass' ) ) {
	class BarClass {
	}
}
```

**Comment by gsherwood** [comment]
Closing due to lack of additional interest for a new sniff
